src/v/raft/timeout_jitter.h

```diff
@@ -11,12 +11,41 @@
 
 #pragma once
 
+#include "config/property.h"
 #include "raft/types.h"
 #include "random/simple_time_jitter.h"
 
 namespace raft {
+class timeout_jitter {
+public:
+    explicit timeout_jitter(config::binding<std::chrono::milliseconds> timeout)
+      : _base_timeout(timeout)
+      , _time_jitter(_base_timeout()) {
+        timeout.watch([this] { update_base_timeout(); });
+    }
+    raft::clock_type::time_point operator()() { return _time_jitter(); }
 
-using timeout_jitter
-  = simple_time_jitter<raft::clock_type, raft::duration_type>;
+    raft::clock_type::duration base_duration() const {
+        return _time_jitter.base_duration();
+    }
+
+    raft::clock_type::duration next_duration() {
+        return _time_jitter.next_duration();
+    }
+
+    raft::clock_type::duration next_jitter_duration() {
+        return _time_jitter.next_jitter_duration();
+    }
+
+private:
+    void update_base_timeout() {
+        _time_jitter
+          = simple_time_jitter<raft::clock_type, raft::duration_type>(
+            _base_timeout());
+    }
+
+    config::binding<std::chrono::milliseconds> _base_timeout;
```

Comment:
I wonder is it okay to have a lot of bindings in many partitions case? Those will have to be updated synchronously

Reply:
good point, we already have one binding per raft instance i.e. disk_timeout, but here it is good point

Reply:
we need to remember here that the number of iterations is bounded by number of partitions per shard, therefore i think we should be fine